File: ruby/lib/svix/models/sink_status.rb
# frozen_string_literal: true
# This file is @generated
module Svix
  class SinkStatus
    ENABLED = "enabled".freeze
    PAUSED = "paused".freeze
    DISABLED = "disabled".freeze
    RETRYING = "retrying".freeze

    def self.all_vars
      @all_vars ||= [ENABLED, PAUSED, DISABLED, RETRYING].freeze
    end

    def initialize(value)
      unless SinkStatus.all_vars.include?(value)
        raise "Invalid ENUM value '#{value}' for class #SinkStatus"
      end

      @value = value
    end

    def self.deserialize(value)
      return value if SinkStatus.all_vars.include?(value)
      raise "Invalid ENUM value '#{value}' for class #SinkStatus"
    end

    def serialize
      @value
    end
  end
end

File: java/lib/src/main/java/com/svix/models/SinkStatus.java
// This file is @generated
package com.svix.models;

import com.fasterxml.jackson.annotation.JsonValue;
import com.svix.Utils.ToQueryParam;

public enum SinkStatus implements ToQueryParam {
    ENABLED("enabled"),
    PAUSED("paused"),
    DISABLED("disabled"),
    RETRYING("retrying");
    private final String value;

    SinkStatus(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}
